### PR TITLE
Fix compile-time test for support of std::make_unique

### PR DESCRIPTION
**Pull request by uklotzde**
I think I found a bug in our adopted util/memory.h by analyzing the VS2013 build errors:

http://builds.mixxx.org:8081/jenkins/job/master/architecture=amd64,platform=windows/378/console

Both of the following files are included and provide an ambiguous version of std::make_unique():
C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\INCLUDE\memory
C:\mixxx\workspace\master\architecture\amd64\platform\windows\src\util/memory.h

I also had to modify the detection for GCC and take the option -std=c++11 into account!

### CODE REVIEW

File: src/util/memory.h [hunk 1]
```diff
@@ -52,13 +52,13 @@
    #elif defined(__GNUC__)
       // std::make_unique was added in gcc 4.9
       #define GCC_VERSION (__GNUC__ * 10000 + __GNUC_MINOR__ * 100 + __GNUC_PATCHLEVEL__)
-      #if GCC_VERSION >= 40900
+      #if GCC_VERSION >= 40900 && __cplusplus > 201103L
```
Comment (rryan) [inline src/util/memory.h]: Why was this needed? Can you please add a comment noting why?

File: src/util/memory.h [hunk 2]
```diff
@@ -33,7 +33,7 @@
 
 // If user hasn't specified COMPILER_SUPPORTS_MAKE_UNIQUE then try to figure out
 // based on compiler version if std::make_unique is provided.
-#if defined(COMPILER_SUPPORTS_MAKE_UNIQUE)
+#if !defined(COMPILER_SUPPORTS_MAKE_UNIQUE)
```
Comment (rryan) [inline src/util/memory.h]: Huh! Whoops. @bstreiff this was what you intended, right?

File: src/util/memory.h (hunk 1 above)
Comment (rryan) [inline src/util/memory.h]: Sorry, missed the mention in the PR description. So this would prevent defining that the compiler supports make_unique when the user specified -std=c++04 for example?

Comment (uklotzde) [inline src/util/memory.h]: Indeed, compiling with -std=c++14 should pull in <memory> from GCC instead 
of this workaround. But I have not tested it.

I've also added an additional commit with a comment why we need to check 
__cplusplus here.

File: src/util/memory.h (hunk 2 above)
Comment (bstreiff) [inline src/util/memory.h]: Whoops indeed. My testing had been more focused around compilers that didn't have std::make_unique and less around the compilers that already did.

In addition, yes, the gcc (and clang?) checks should also check __cplusplus in order to differentiate between -std=c++98 or c++11 or c++14.

Comment (rryan) [inline src/util/memory.h]: Great, thanks!
